cccatalog-api/test/v1_integration_test.py

```diff
@@ -0,0 +1,502 @@
+            'HOST': '127.0.0.1',
+            'NAME': 'openledger',
+            'PASSWORD': 'deploy',
+            'USER': 'deploy',
+            'PORT': 5432
+        }
+
+
+@pytest.mark.django_db
+def test_oauth2_verification(test_oauth2_token, django_db_setup):
+        # This test needs to cheat by looking in the database, so it will be
+        # skipped in non-local environments.
+        if API_URL == 'http://localhost:8000':
+            _id = OAuth2Verification.objects.aggregate(Max('id'))['id__max']
+            verify = OAuth2Verification.objects.get(id=_id)
+            code = verify.code
+            url = f'{API_URL}/v1/auth_tokens/verify/{code}'
+            response = requests.get(url)
+            assert response.status_code == 200
+            test_oauth2_rate_limit_reporting(test_oauth2_token, verified=True)
+
+
+def test_watermark_preserves_exif():
+    img_with_exif = 'https://raw.githubusercontent.com/ianare/exif-samples/' \
+                    'master/jpg/Canon_PowerShot_S40.jpg'
+    info = {
+        'title': 'test',
+        'creator': 'test',
+        'license': 'test',
+        'license_version': 'test'
+    }
+    _, exif = watermark(image_url=img_with_exif, info=info)
+    assert exif is not None
+
+    img_no_exif = 'https://creativecommons.org/wp-content/uploads/' \
+                  '2019/03/9467312978_64cd5d2f3b_z.jpg'
+    _, no_exif = watermark(image_url=img_no_exif, info=info)
+    assert no_exif is None
+
+
+def test_attribution():
+    """
+    The API includes an attribution string. Since there are some works where
+    the title or creator is not known, the format of the attribution string
+    can need to be tweaked slightly.
+    """
+    title_and_creator_missing = Image(
+        identifier="ab80dbe1-414c-4ee8-9543-f9599312aeb8",
+        title=None,
+        creator=None,
+        license="by",
+        license_version="3.0"
+    )
+    assert "This work" in title_and_creator_missing.attribution
+
+    title = "A foo walks into a bar"
+    creator_missing = Image(
+        identifier="ab80dbe1-414c-4ee8-9543-f9599312aeb8",
+        title=title,
+        creator=None,
+        license="by",
+        license_version="3.0"
+    )
+    assert title in creator_missing.attribution
+    assert "by " not in creator_missing.attribution
+
+    creator = "John Doe"
+    title_missing = Image(
+        identifier="ab80dbe1-414c-4ee8-9543-f9599312aeb8",
+        title=None,
+        creator=creator,
+        license="by",
+        license_version="3.0"
+    )
+    assert creator in title_missing.attribution
+    assert "This work" in title_missing.attribution
+
+    all_data_present = Image(
+        identifier="ab80dbe1-414c-4ee8-9543-f9599312aeb8",
+        title=title,
+        creator=creator,
+        license="by",
+        license_version="3.0"
+    )
+    assert title in all_data_present.attribution
+    assert creator in all_data_present.attribution
+
+
+def test_browse_by_provider():
```

Comment:
I think it would be worth renaming tests like these to more generic names since we don't have anything like "browse by provider" in the code anymore and we call them sources now anyway.

Reply:
Please look through the other tests for items like these, I haven't taken a comprehensive look.

Reply:
I renamed some test cases.